File: 프로그래머스/1/176963. 추억 점수/추억 점수.py
def solution(name, yearning, photo):
    yearning_dict = dict(zip(name, yearning))
    result = [0] * len(photo)
    
    for idx, lst in enumerate(photo):
        for name in lst:
            result[idx] += yearning_dict.get(name, 0)
            
    return result
            
            
    

File: 프로그래머스/1/176963. 추억 점수/추억 점수.js
const solution = (name, yearning, photo) => {
    const result = Array(photo.length).fill(0);
    const yearningObject = {};
    name.forEach((name, idx) => yearningObject[name] = yearning[idx]);
    
    photo.forEach((lst, idx) => {
        lst.forEach((name) => {
            result[idx] += yearningObject[name] ?? 0
        });
    });
    return result;
}